src/macaron/slsa_analyzer/analyzer.py

```diff
@@ -836,6 +841,52 @@ def perform_checks(self, analyze_ctx: AnalyzeContext) -> dict[str, CheckResult]:
 
         return results
 
+    def _resolve_more_dependencies(
+        self, dependencies: dict[str, DependencyInfo], context: AnalyzeContext, description: str
+    ) -> None:
+        """Utilise the Repo Finder to resolve more dependencies to repositories."""
+        purl = PackageURL.from_string(context.component.purl)
+        build_tools = context.dynamic_data["build_spec"]["tools"]
+        if not purl.type or not build_tools:
+            logger.debug("No PURL type found for: %s", purl)
+        else:
+            # Get actual PURL type based on current build tool
+            build_tool = build_tools[0]
```

Comment:
What is the reason we select the first element in `build_tools` to resolve the dependencies. Would it prevent us from acquiring the correct dependencies if a component have more than one detected build tool?

Reply:
You are right, there would be an issue with multiple conflicting build tools. One solution I see for this is to add the name of the tool each dependency was found by. Perhaps in the `note` field of `DependencyInfo`?

Reply:
I think by refactoring the repo finder differently we can avoid this issue. Instead of moving `_resolve_more_dependencies` here, we can add it as an abstract method to `BaseBuildTool`, which can be implemented per build type. Note how we get the [dependency analyzer](https://github.com/oracle/macaron/blob/66f138bd882507afbe8cf854e997b47b97029086/src/macaron/slsa_analyzer/analyzer.py#L275) per build tool. We can subsequently call `resolve_more_dependencies` per build tool afterwards.